ssha/aws.py

```diff
@@ -1,19 +1,43 @@
 from __future__ import print_function
 
+from functools import wraps
+
 import boto3
 import boto3_session_cache
 import botocore.session
+from botocore.exceptions import ClientError, ParamValidationError
 
 from . import config
 
 
 _sessions = {}
 
 
+def retry(attempts=3):
+    def wrapper(f):
+        @wraps(f)
+        def wrapped(*args, **kwargs):
+            tries = attempts
+            while tries:
+                tries -= 1
+                try:
+                    return f(*args, **kwargs)
+                except (ClientError, ParamValidationError) as error:
+                    if tries > 0:
+                        print(error)
+                    else:
+                        raise
+            return f(*args, **kwargs)
```

Comment:
I don't think this line will ever run, unless you used `retry(attempts=0)`. You could delete this line and change `while tries:` to `while True:` and I think it'd do the same thing.

Also I'm not a fan of single letter variables. I think `func` is commonly used in decorators, and it's what I would use.